build-image-src/Dockerfile-python311

```diff
@@ -0,0 +1,53 @@
+ARG IMAGE_ARCH
+FROM public.ecr.aws/lambda/python:3.11-$IMAGE_ARCH
+
+RUN yum groupinstall -y development && \
+  yum install -d1 -y \
+  yum \
+  tar \
+  gzip \
+  unzip \
+  python3 \
+  jq \
+  grep \
+  curl \
+  make \
+  rsync \
+  binutils \
+  gcc-c++ \
+  procps \
+  libgmp3-dev \
+  zlib1g-dev \
+  libmpc-devel \
+  python3-devel \
+  && yum clean all
+
+# Install AWS CLI
+ARG AWS_CLI_ARCH
+RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-$AWS_CLI_ARCH.zip" -o "awscliv2.zip" && unzip awscliv2.zip && ./aws/install && rm awscliv2.zip && rm -rf ./aws
+
+# Install SAM CLI in a dedicated Python virtualenv
+# Note: Remove cython<3.0.0 line once the issue with building pyyaml from source is fixed
+ARG SAM_CLI_VERSION
+RUN curl -L "https://github.com/awslabs/aws-sam-cli/archive/v$SAM_CLI_VERSION.zip" -o "samcli.zip" && \
+  unzip samcli.zip && python3 -m venv /usr/local/opt/sam-cli && \
+  /usr/local/opt/sam-cli/bin/pip3 install wheel && \
+  /usr/local/opt/sam-cli/bin/pip3 install "cython<3.0.0" pyyaml==5.4.1 --no-build-isolation && \
+  /usr/local/opt/sam-cli/bin/pip3 uninstall wheel cython --yes && \
```

Comment:
nit: can we move this line up so that cleanup will be removal of single line?

Reply:
done